File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, f: List[int], n: int) -> bool:
        count = 0
        for i in range(len(f)):
            if f[i] == 0:
                left = (i == 0) or (f[i-1] == 0)
                right = (i == len(f)-1) or (f[i + 1] == 0)
                if left and right:
                    f[i] = 1
                    count += 1
        return count >= n

File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] f, int n) {
        for (int i = 0; i < f.length; i++) {
            if (f[i] == 0) {
                boolean left = i == 0 || f[i-1] == 0;
                boolean right = i == f.length-1 || f[i+1] == 0; 
                if (left && right) {
                    f[i] = 1;
                    n--;
                }
            }
        }
        return n < 1;
    }
}